### ISSUE TITLE
Diffusion Post-Training issue: Loss showed as negative values, Data loaded for first batch showed as static frames, keys mismatched after loading post-trained models

### ISSUE THREAD

**Issue by Rxliang** [nvidia-cosmos/cosmos-predict1#56]
Two lr settings have been tried with both lr=0 and lr=4.957595192603974e-05 as set by default with the demo codes.
Then first output-batch x0 has been visualized by saving its tensor and reloaded for the inference world pipeline showed static frames. torch.Size([1, 16, 16, 90, 160])
https://github.com/user-attachments/assets/363181b5-9386-4e60-b8ad-f49d001b3bc5
several keys missing with extra_state skipped due which should be fine with transformer_engine implementation yet concerning therefore still posted here FYI "_IncompatibleKeys(missing_keys=[], unexpected_keys=['net.extra_pos_embedder.pos_emb_h', 'net.extra_pos_embedder.pos_emb_w', 'net.extra_pos_embedder.pos_emb_t'], incorrect_shapes=[])"
Therefore current post-trained model failed to generate video but only noisy frames. 
https://github.com/user-attachments/assets/f54bc1b6-e2e6-4d09-8cb3-b6a2801f3696

Thanks in advance for your atttention and time for checking these.

[stdout_4.957595192603974e-05.log](https://github.com/user-attachments/files/19740270/stdout_4.957595192603974e-05.log)
[stdout_lr=0.log](https://github.com/user-attachments/files/19740269/stdout_lr.0.log)

**Comment by Rxliang** [comment]
And ema model was extremely big 26GB which is about double the size of the reg model. I am wondering if there is any explanation for this.

**Comment by KiK0S** [comment]
I am observing (kind of) the same behavior trying to finetune video2world model. Grid-like artefacts are really similar and negative loss is also there for me. Haven't tried the first output-batch.

Based on the loss function (`model_image.py` + section 5.1 of the paper) I assume disabling `loss_add_logvar` in config should force non-negativity. When this flag is enabled, it's difficult to reason about the loss value.

**Comment by qsh-zh** [comment]
we are using final loss = score matching loss + entropy regularizer, originally introduced in edm2 paper. 

if you want to log the score matching loss~(also known as velocity prediction, noise prediction), we can log "mse_loss" or "edm_loss" [the line](https://github.com/nvidia-cosmos/cosmos-predict1/blob/main/cosmos_predict1/diffusion/training/models/model_image.py#L238-L249), which is guaranteed to be positive.

**Comment by qsh-zh** [comment]
regarding ema model and regular model size. the regular model is in bf16 tensor dtype while by default ema model is in fp32.  users can change ema dtype to bf16, which should also work.

**Comment by Rxliang** [comment]
Thanks for your reply however the lr=0 is supposed to be essentially the pre-trained model right however after post-training it generated noise-only video. 
What should I do to avoid this? 
Is that something to do with the model loading during the post-training process like the mismatch keys listed in my original ask?

**Comment by qsh-zh** [comment]
the config looks problematic to me. `extra_per_block_abs_pos_emb` should be true.

**Comment by Rxliang** [comment]
https://github.com/nvidia-cosmos/cosmos-predict1/blob/cae01cfebc3dc76bd522a6bd301a20d03af678ba/cosmos_predict1/diffusion/training/config/text2world/experiment.py#L255
Here only in text2world it was declared explicitly True but not in Video2world. 
https://github.com/nvidia-cosmos/cosmos-predict1/blob/cae01cfebc3dc76bd522a6bd301a20d03af678ba/cosmos_predict1/diffusion/training/networks/general_dit.py#L181
And here in the network, it was inherently turned to False. Should I copy experiment setting in T2W to V2W in order to make it work?

**Comment by qsh-zh** [comment]
I believe it should be True.

**Comment by KiK0S** [comment]
I did the other way round: I put false in the inference mode instead of true for the training. after some finetuning it started working. So yes, seems like config mismatch.

The same issue is also there for multi-view post-training setup, but I haven't found a cause yet.
